ctapipe/core/tool.py

```diff
@@ -202,23 +206,28 @@ def run(self, argv=None):
             Provenance().finish_activity(activity_name=self.name)
         except ToolConfigurationError as err:
             self.log.error(f'{err}.  Use --help for more info')
+            return 2  # wrong cmd line parameter
```

Comment:
maybe using `sys.exit` everywhere here in `Tool.run` would be the simplest solution.

 The tests could then do something like this:
```python
try:
    tool.run()
except SystemExit as e:
    assert e.code == 0
```

Reply:
or just put the "return" in the `finally:` block, and make a variable like `exit_code=2`, since we don't really want to exit exactly inside that except block (things in finally should still run)

Reply:
but using `sys.exit()` is also fine, as long as it happens later.

Reply:
As long as the `finally:` block still runs, I think that is ok, and is probably more clear. I didn't realize `sys.exit()` raises an exception rather than immediately exists (as in c)!

Reply:
The goal being to also automatically store the exit information in the provenance (returning immediately, breaks that)

Reply:
That should probably solver by using provenance as context Manager und in run?

Exit geht's the exception und ist automatically reraised if Exit does not return true

Reply:
yes, i agree - making provenance a context manager would be much nicer. I want to clean it up anyhow (the current interface is not great).  For this PR, i think it should still work, since we still get to the finally block if there is an error.